bindings/experimental/distrdf/python/DistRDF/Backends/Spark/Backend.py

```diff
@@ -149,6 +149,15 @@ def distribute_unique_paths(self, paths):
             self.sc.addFile(filepath)
 
     def make_dataframe(self, *args, **kwargs):
-        """Creates an instance of SparkDataFrame"""
-        headnode = HeadNode.get_headnode(*args)
-        return DataFrame.RDataFrame(headnode, self, **kwargs)
+        """
+        Creates an instance of distributed RDataFrame that can send computations
+        to a Spark cluster.
+        """
+        # Set the number of partitions for this dataframe, one of the following:
+        # 1. User-supplied `npartitions` optional argument
+        # 2. An educated guess according to the backend, using the backend's
+        #    `optimize_npartitions` function
+        # 3. Set `npartitions` to 2
+        npartitions = kwargs.pop("npartitions", self.optimize_npartitions(Base.BaseBackend.MIN_NPARTITIONS))
```

Comment:
no need to pass `MIN_NPARTITIONS` to a method, it's a data member

Reply:
Right, initially it was meant to be a generic function "optimize the partitions if possible, else give me back the argument I pass you". It's probably changed logic now

Reply:
So what should I do here? Change the signature of `optimize_npartitions` to `(self)` rather than `(self, npartitions)` ? Or maybe leave the input parameter but make it optional as `(self, npartitions=None)`?

Reply:
(this function might still change / be removed in the future whenever we decide on some coherent structure to "optimize" the number of partitions of the distributed dataframe)

Reply:
i don't see why this shouldn't be `def optimize_npartitions(self)`